### ISSUE TITLE
Not getting data for the DCC Sourced smart gas meter Usage (today)

### ISSUE THREAD

**Issue by daknightuk** [HandyHat/ha-hildebrandglow-dcc#311]
**Describe the bug**
I am not getting data for the DCC Sourced smart gas meter Usage (today)

**To Reproduce**
I removed the entire integration from HACS, restarted my RPI4+
I added the integration through HACS
I restarted my RPI4+
I checked that the entities were shown 
![image](https://user-images.githubusercontent.com/84586585/219381409-40255843-5e87-4679-b951-84cc8eca5195.png)

**Expected behaviour**
I was expecting to see gas usage in my energy dashboard as I am using gas and the Bright App shows I do have usage.
I am getting an error in the gas consumption under the Energy Dashboard settings
This seemed to happen after my gas and electricity meter stopped reporting gas usage for about 24 hours then appeared to be "catching up"

**Screenshots**
![image](https://user-images.githubusercontent.com/84586585/219381118-e3272412-410c-4bc9-b576-b67c9fd24672.png)
![image](https://user-images.githubusercontent.com/84586585/219381685-93bc614e-5f49-4248-a37a-46bfd775b831.png)
![image](https://user-images.githubusercontent.com/84586585/219381903-545281e5-c292-4a8a-8b77-e683f2664bc4.png)

**Version**
Integration 1.0.3
Home Assistant 2023.2.5
Supervisor 2023.01.1
Operating System 9.5
Frontend 20230202.0 - latest

**Debug log**
Log Details (WARNING)
Logger: homeassistant.components.sensor
Source: components/sensor/__init__.py:503
Integration: Sensor ([documentation](https://www.home-assistant.io/integrations/sensor), [issues](https://github.com/home-assistant/home-assistant/issues?q=is%3Aissue+is%3Aopen+label%3A%22integration%3A+sensor%22))
First occurred: 13:53:07 (2 occurrences)
Last logged: 13:53:07

Entity sensor.dcc_sourced_smart_electricity_meter_cost_today (<class 'custom_components.hildebrandglow_dcc.sensor.Cost'>) is using state class 'total_increasing' which is impossible considering device class ('monetary') it is using; expected None or one of 'total'; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.
Entity sensor.dcc_sourced_smart_gas_meter_cost_today (<class 'custom_components.hildebrandglow_dcc.sensor.Cost'>) is using state class 'total_increasing' which is impossible considering device class ('monetary') it is using; expected None or one of 'total'; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.

**Additional context**
I don't know why it doesn't work I hope the above helps you fix it

**Comment by daknightuk** [comment]
@adam-cobb do you mean you are NOT getting gas usage data? do you have the bright app also. I can see my hourly usage on the bright app although the update is "lagging" behind by a while.

**Comment by tekhiegit** [comment]
Same no usage data all week, before then it was missing large quantities of daily usage too.

**Comment by adam-cobb** [comment]
Yeah sorry my bad on the typo. I haven't been getting any gas usage! Looks fine in the Bright app for me also.

**Comment by daknightuk** [comment]
I think this MAY be the same as the issue "Data appears incomplete when compared to other DCC fed sources. #305" but no-one has reported the error which I'm getting which is odd. Having said that my energy provider e.on does tend to "faff" around and break things that worked fine before (on my 2nd set of smart meters now) so not sure if its an issue at their end.

**Comment by chriskchambers** [comment]
I would also like to say that, from last Thursday I had no data from DCC Sourced smart gas meter Plug in, into my Home Assistant, but when I go to the Phone App, I can see my data.

**Comment by stewartdoyle** [comment]
No gas data within the Hilderbrand-DCC integration since 9 Feb for me too. Electricity usage OK for DCC and gas and electricity are both fine in the Bright app. I am with Bulb and wonder if it might be related to move to Octopus?

**Comment by Eddie46848** [comment]
The app is updating once a day now and this integration is missing the data, so isn't recording anything.
I don't think it's the integrations fault, the old saying is you don't get anything for free and glowmarkt are trying to sell a product.

**Comment by southwalesboy** [comment]
Same here, gas usage data stopped reporting on February 8th, but the data is showing in the 'Bright' app on my smartphone.

**Comment by HandyHat** [comment]
Hey all, this looks like it may be an extension of the issue described in #305, as the root issue seems to be the data being delayed over a day, resulting in no data being collected for that day.

Feel free to follow the discussion there!

**Comment by nigam-sa** [comment]
I too noticed the same issue. No data for gas available in HA since 10th Feb. Electric data is fine. 
In the bright app i can see the gas data.
<img width="1012" alt="image" src="https://user-images.githubusercontent.com/59933187/220209943-e35ca6b9-4a74-4057-88a2-12ec3864ff42.png">
<img width="1008" alt="image" src="https://user-images.githubusercontent.com/59933187/220209981-99fc5057-6717-4da2-82fb-c8d7f600fbb9.png">

**Comment by ethzero** [comment]
Really wish I'd reported my glitches back on the 13/14th, but in fairness wanted to see if others were getting the same issue.

Gas reading starting glitching on the 13th Feb and lost data for most of that day. Data seemed to come back on the 14th but was massively inaccurate. Then stopped again for over a day on the 16th and since then I've been getting a bizarrely low reading again completely contrary according to my IHD.

Electricity readings _appear_ to be coming in but again are completely inaccurate. Up until the 14th you can clearly see the rise in energy usage around the every, when my gaming PC is in use however all days since then have falsely show negligibly use, with the weekend of the 18th/19th not showing the practically 48 hour use :) 

![image](https://user-images.githubusercontent.com/357538/220220949-3375d1b9-d2b4-49e5-88e8-dd4598c174e3.png)
![image](https://user-images.githubusercontent.com/357538/220220963-e259d8ad-4e68-4af9-9e65-a9023d9c9b4b.png)

**Comment by ethzero** [comment]
> I think this MAY be the same as the issue "Data appears incomplete when compared to other DCC fed sources. #305" but no-one has reported the error which I'm getting which is odd. Having said that my energy provider e.on does tend to "faff" around and break things that worked fine before (on my 2nd set of smart meters now) so not sure if its an issue at their end.

I'm with e.on and I've found HA+hildebrandglow to be pretty solid since I set this up last December and only had one other couple-hour glitch in all that time back in January. The latest "glitch" has resulted in a feed for me that seems both intermittently broken and what data does get through it woefully inaccurate. As others have noted I do appear to be getting good data on the app *for the previous 24 hours* but the entire *current day* is blank  for both electricity and gas.

**Comment by m24andrew** [comment]
I too have just noticed to day that I have no gas consumption data since the 14th Feb, all fine on Bright app.

**Comment by cliveeisen** [comment]
This implies your software is making a different api call to get the consumption data than bright does.
If someone tells us exactly what call is not working compared to bright we will have a look.
Ideally please raise a ticket on this specific issue by emailing support@glowmarkt.com
The reality is if you want realtime up to date data buy the glowmarkt IHD and get a local mqtt client - everything else is something of a fudge.

**Comment by chriskchambers** [comment]
I can't reminder the procedure I did to get my Gas reading into HA, but looking around my environment, I see most of the steps here has been completed. https://github.com/HandyHat/ha-hildebrandglow-dcc/blob/main/README.md

Installation
Automated installation through HACS
You can install this component through [HACS](https://hacs.xyz/) to easily receive updates. Once HACS is installed, click this link:

Open your Home Assistant instance and open a repository inside the Home Assistant Community Store.

Manually add to HACS
Manual installation
Copy the custom_components/hildebrandglow_dcc/ directory and all of its files to your config/custom_components/ directory.

Configuration
Once installed, restart Home Assistant:

Open your Home Assistant instance and show the system dashboard.

Then, add the integration:

Open your Home Assistant instance and start setting up a new integration.

Manually add the Integration
Sensors

**Comment by JohnMcLear** [comment]
Mine stopped working on the 19th of Feb, data is available in Bright app but not showing from integration.  Thanks!

**Comment by thorrrr** [comment]
HI
Last time i had a full day readings was 22 Jan 2023 , both Elec & Gas.
Bright App full of data

**Comment by Gazoonky** [comment]
Similar here. Problems with data in HA for several weeks but nothing last 2 weeks. Switched from British Gas to Octopus and this is what I thought had caused the issue but there seems to be something else wrong. Bright app reporting data normally

**Comment by AKruimink** [comment]
Set mine up for the first time yesterday, the mobile app seems to give me data just fine (is a day behind in data it seems) but in HA both electricity and gas don;t work and give me the Statics_not_defined and Entity unavailable warnings with no data

**Comment by XcOM987** [comment]
TO confuse things a little, this is suspected to be related to #305, but when I login to my bright app it has data from this morning, although I think it's only showing the standing charge so far on the app but will keep an eye on it, but doesn't seem to be feeding in to HA, I get everything but the usage, so I get the rates and standing charge though so it is pulling some data.

What can I do to help investigate this, I have my setup in a VM so I can mess about with it and revert it if something breaks

**Comment by miczlo** [comment]
I'm seeing same behaviour. In the app I can't see data for today only for previous days but in the integration I have no historical data either.

**Comment by planetming** [comment]
My HA seems to have started receiving data again from Hilderbrandglow since 3rd March 2023. Appears to be accurate.
No intervention at all on my part.

**Comment by XcOM987** [comment]
Likewise but unclear how accurate it is since then, I do notice on the 3rd I had a major correction of about negative 1000KWh on the 3rd and since then seems to be mostly OK, I assume based on what I was seeing the issue was with Glow and not the integration and they did some sort of reset to fix something?

**Comment by craigrouse** [comment]
@XcOM987 I had the same "correction", though to the tune of over 3000kWh. A little annoying, since it'll put the measurement out significantly over the year. Glad it wasn't just me anyway! My readings are now matching the Bright app exactly, so I think it's safe to say it's fixed (for now, at least). Worth noting that I only use it for gas, not electricity.

**Comment by CJDumbleton** [comment]
My data still doesn't tally between the HA energy dashboard and the Bright app. 

The total daily usages match (slightly out for electricity). For 6 March:
- Electricity: HA total daily = 7.31 kWh; Bright total daily = 7.372.
- Gas: HA total daily = 51.19 kWh; Bright total daily = 51.1877.

But, the hourly bar charts look different. The HA energy dashboard seems normally to be showing two-hourly bars with some missing.

Bright electricity bar chart
![image](https://user-images.githubusercontent.com/55947036/223413808-b2d24fbd-20a7-4236-8819-4abbd8dc80cd.jpeg)

HA electricity bar chart
![image](https://user-images.githubusercontent.com/55947036/223413853-b6573fbf-27c7-48f7-81ff-8d72b6e52f46.jpeg)

Bright gas bar chart
![image](https://user-images.githubusercontent.com/55947036/223413882-c435b67b-d15e-4520-9ac6-bd134918eb52.jpeg)

HA gas bar chart
![image](https://user-images.githubusercontent.com/55947036/223413931-a945719b-3bb1-4b05-9b6f-ab754217649a.jpeg)

**Comment by daknightuk** [comment]
@lowpeter I went down the same route as you and lost the historical data. Im going to take a guess that you might be able to leave the DCC Sourced smart electricity entity in place for the energy dashboard but then maybe set it to disabled as an entity,
The way I went about sorting mine was to use the utility meter integration, created an import meter and an export meter and then used the solar as the source. You can then calibrate the meters to incorporate a historical usage figure (accumalative)

**Comment by daknightuk** [comment]
> Many thanks for your response daknightuk. This was discouraging, but I have disabled the 'DCC Sourced smart electricity meter Usage (today)' and the historic data remains in the Energy Dashboard. The config looks un-neat as I have a warning that:
> 
> Entity not defined Check the integration or your configuration that provides: sensor.electric_consumption_today
> 
> I can live with this.
> 
> I added the Energy from Grid from the Solar api, but unfortunately this backwards populated for as long as I set up the api and so I have double electric for some months. I hoped it would just populate from the day I added it to the Energy Dashboard.
> 
> I do not want to estimate past electric use as you have done as I wanted to compare current use to past use. Seems that the Solar Api and the DCC are not reporting the same (but similar amounts) for the doubling period. This may be as good as it can be unless there is other suggestions. Thanks again.

Try putting the data into a spreadsheet and see what the differential is, maybe look at "templating" it (i've not done this yet)
The Solar system will always be different to the actual meter because it uses a CT clamp which measures the electricity flow through the cable. The actual meter has a physical connection to the cable and as thats what we are "billed for" I would go with that if looking at accuracy.

**Comment by chriskchambers** [comment]
I am sad.to report that this has any started to failed.to.update my HA dashboard.  But the app on the phone is still working.

HA stopped working 2 days ago.  On the 28/4

**Comment by daknightuk** [comment]
@chriskchambers I got fed up of the unreliablity of Hildebrand, ok so its a free service and it has a lot of demand but it seems to break all the time. I only have it in HA now as a backup as I signed up to Octopus who offer an API so I can get my gas data directly from them. I'm sure there will be other companies offering API's also.

**Comment by craigrouse** [comment]
The problem is that the Octopus data is 1 day behind, and only gives you the total daily figure. That's why this integration is so useful. I'd even pay Hildebrand a small amount each year for API-only access if it worked.

**Comment by daknightuk** [comment]
@craigrouse Agreed it is a bit behind but lets be honest if its cold I put the heating on and if its warm I'll turn it off. I don't see any cost saving to be achieved by tracking it any closer than that.

**Comment by abjwsm** [comment]
Hi I also have the same issue. Several weeks now. I have Electric but no gas data in both the phone app and Home assistant. It was working and I have changed nothing. I contacted Hildebrand & GlowMarket and they say they can see my Gas data at their end. I am using British Gas SMETS2 but the display unit has no built in WiFi. Are there any other methods available? Al

**Comment by Bookermind** [comment]
I have the same issue (only just installed this integration).
However when I look into the logs I can see the following:
`2023-08-13 20:23:34.321 WARNING (MainThread) [homeassistant.components.sensor] Entity sensor.dcc_sourced_smart_electricity_meter_cost_today (<class 'custom_components.hildebrandglow_dcc.sensor.Cost'>) is using state class 'total_increasing' which is impossible considering device class ('monetary') it is using; expected None or one of 'total'; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.`

**Comment by craigrouse** [comment]
For any Octopus customers having issues with this integration, you should request an Octopus Home Mini from them, as that now reports gas usage via the regular Octopus API hourly, and importing to HA is handled automatically by the [Octopus Energy HA component](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy). It works brilliantly and allowed me to stop using the Hildebrand Glow integration.

**Comment by plord12** [comment]
> For any Octopus customers having issues with this integration, you should request an Octopus Home Mini from them, as that now reports gas usage via the regular Octopus API hourly, and importing to HA is handled automatically by the [Octopus Energy HA component](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy). It works brilliantly and allowed me to stop using the Hildebrand Glow integration.

I've requested one but seems they don't have stock :-(

**Comment by jbsugden** [comment]
I have this exact issue.  Any update on a fix?

**Comment by chrisdean101** [comment]
Same issue here! With the same error code as above. Does anyone know what the root course is.

**Comment by adamsfarley** [comment]
Has there been a fix for this?
